File: apps/minecraft/spigot/src/main/java/crabcraft/net/crabUtilities/jade/protocol/provider/ServerDataProvider.java
package crabcraft.net.crabUtilities.jade.protocol.provider;

import net.minecraft.nbt.CompoundTag;
import crabcraft.net.crabUtilities.jade.protocol.accessor.Accessor;

public interface ServerDataProvider<T extends Accessor<?>> extends JadeProvider {
    void appendServerData(CompoundTag data, T accessor);
}

File: apps/minecraft/spigot/src/main/kotlin/crabcraft/net/crabUtilities/jade/protocol/provider/ServerDataProvider.kt
package crabcraft.net.crabUtilities.jade.protocol.provider

import crabcraft.net.crabUtilities.jade.protocol.accessor.Accessor
import net.minecraft.nbt.CompoundTag

interface ServerDataProvider<T : Accessor<*>> : JadeProvider {
    fun appendServerData(data: CompoundTag, accessor: T)
}
